IANA timezone

Allcharles opened the pull request:
# IANA Timezone Input

Created an IANA compatible timezone input. When baw-server is updated to allow setting of timezones, this should be compatible.

## Problems

- No unit tests, attempted but gave up for now
  - Will need to verify outputs are correct when baw-server is fixed
  - Issues may be spectator related, might have to write tests using default library
- Changes break site, register, and edit profile forms
  - Will be addressed in another PR which merges into this one

## Issues

Related: #51  
Related: https://github.com/QutEcoacoustics/baw-server/issues/410

## Visual Changes

![image](https://user-images.githubusercontent.com/3955116/88355005-38d57580-cda6-11ea-9dbb-9eeffe1c4624.png)

![image](https://user-images.githubusercontent.com/3955116/88355039-5acef800-cda6-11ea-8855-4680ec68cc30.png)

![image](https://user-images.githubusercontent.com/3955116/88355013-41c64700-cda6-11ea-9e8d-7c698e6d85e6.png)

![image](https://user-images.githubusercontent.com/3955116/88355022-48ed5500-cda6-11ea-852c-9f1c047ad7f5.png)

Allcharles committed:
Working timezone displays

Allcharles committed:
Cleanup input formatting

Allcharles committed:
FormlyTimezoneInput unit test boilerplate

Allcharles committed:
Linted project

Allcharles committed:
Working timezone input using new dep

Replaced moment-timezone library with @vvo/tzdb

atruskie commented on src/app/component/shared/formly/timezone-input.component.ts:
```diff
@@ -1,67 +1,133 @@
+        [editable]="false"
+        [ngbTypeahead]="search"
+        [inputFormatter]="formatter"
+        [resultTemplate]="rt"
```
We can choose a better name than `rt`

atruskie commented on src/app/component/shared/formly/timezone-input.component.ts:
```diff
@@ -1,67 +1,133 @@
+        <ngb-highlight
+          [result]="r.currentTimeFormat"
+          [term]="t"
+        ></ngb-highlight>
```
the screenshot in the PR shows the textual labels using the `name` value instead of `currentTimeFormat` - is that an issue with the screenshot or is this code not working as intended?

Allcharles replied:
Screenshot, i forgot to update it, my bad

Allcharles committed:
Fixed review comments

Allcharles committed:
Custom Form Input Cleanup (#354)

* Working site location and timezone fields

* Removed fieldGroups

- Removed all fieldGroups
- Created a couple files to handle location ahd password verifications
- Make custom-inputs module to group custom formly inputs

* Cleanup how render-field handles images

* Created password confirmation input

* Added map to location-input

* Change Detection Loop Bug

Fixed a bug where infinite change detection would occur on the `MapComponent`. This stemmed from the `Site.getMapMarker()` function which returns a new object each time it is called. If the function is called directly into a components HTML, change detection will continue to evaluate it as a new object and infinitely load eventually lagging out the page. To solve, I've created a `sanitizeMapMarkers()` function which can be used to fix change detection for single, and multiple marker use cases.

* Added visual feedback to timezone input

Added country name to give user visual feedback when a timezone has been selected for a form. Also extracted the IndicatorComponent so that it can be used as a module.

* Fix unit tests (#362)

* Cleaned up broken unit tests

Had an edge case where it appeared that I could not mock the MapComponent. This turns out to stem from the fact that unit tests cannot override the components for modules which are not directly tied to the component. In this case, the custom formly inputs are loaded as part of a smaller module, which means I cannot override the MapComponent

* Mock services (#363)

* Extract service/guard to modules

* Removed testAppInitializer

* Removed testBawServices

* Mocked services with SpyObjects

* Created ApiErrorDetails generator

Allcharles committed:
Merge branch 'master' into iana-timezone

Allcharles committed:
Fixed broken orphan filter reference

Allcharles committed:
Update deps

Allcharles committed:
Update package-lock.json

Allcharles committed:
Fix typing errors

Allcharles committed:
Re-added unit tests to CI

Allcharles committed:
Removed ProjectsRequestComponent tests

Tests are failing for some reason, currently they dont test anything so its no shame to remove

Allcharles committed:
Removed wrong test
